tests/test_process.py

```diff
@@ -30,3 +55,141 @@ def test_process_inputs(generate_calc_job_node):
     selected_input = orm.load_node(uuid)
 
     assert process_input_widget.info.value == f"PK: {selected_input.pk}"
+
+
+@pytest.mark.usefixtures("aiida_profile_clean")
+def test_process_outputs_widget(multiply_add_completed_workchain):
+    """Test ProcessOutputWidget with a simple `WorkChainNode`"""
+    from aiidalab_widgets_base.process import ProcessOutputsWidget
+
+    # Test the widget can be instantiated with empty inputs
+    widget = ProcessOutputsWidget()
+
+    # Test the widget can be instantiated with a process
+    widget = ProcessOutputsWidget(process=multiply_add_completed_workchain)
+
+    # Simulate output selection.
+    widget.show_selected_output(change={"new": "result"})
+
+
+@pytest.mark.usefixtures("aiida_profile_clean")
+def test_process_follower_widget(multiply_add_process_builder_ready):
+    """Test ProcessFollowerWidget with a simple `WorkChainNode`"""
+    from aiidalab_widgets_base.process import ProcessFollowerWidget
+
+    # Test the widget can be instantiated with empty inputs
+    widget = ProcessFollowerWidget()
+
+    process = engine.submit(multiply_add_process_builder_ready)
+
+    process.seal()
+
+    # Test the widget can be instantiated with a process
+    widget = ProcessFollowerWidget(process=process)
+
+    # Follow the process till it is completed.
+    widget.follow()
+
+
+@pytest.mark.usefixtures("aiida_profile_clean")
+def test_process_report_widget(multiply_add_completed_workchain):
+    """Test ProcessReportWidget with a simple `WorkChainNode`"""
+    from aiidalab_widgets_base.process import ProcessReportWidget
+
+    # Test the widget can be instantiated with empty inputs
+    widget = ProcessReportWidget()
+
+    # Test the widget can be instantiated with a process
+    widget = ProcessReportWidget(process=multiply_add_completed_workchain)
+    widget.update()
```

Comment:
Is that possible to test some tangible thing happened here? The same for the tests below, I think just running `update` is sort of meaningless.

Reply:
Right, thanks. I fixed that in [7423f60](https://github.com/aiidalab/aiidalab-widgets-base/pull/455/commits/7423f60bf1e436c6a34d75263cfe31f3c9d4d113) 👍

Reply:
Sorry, you misunderstand me here. I was trying to ask you to add an assert clause to test the widget is changed after the `update` is called. The widget value should be update in terms of the process information.

For instance, the `ProcessReportWidget` here, after call `widget.update()`, the `self.value` is updated and it will contain some information about the input process.

Reply:
Ok, done. I changed the other widgets, however. The process doesn't produce a report here, so the message stays the same.